superset/connectors/sqla/models.py

```diff
@@ -529,11 +529,7 @@ def get_sqla_query(  # sqla
             inner_select_exprs += [inner_main_metric_expr]
             subq = select(inner_select_exprs)
             subq = subq.select_from(tbl)
-            inner_time_filter = dttm_col.get_time_filter(
-                inner_from_dttm or from_dttm,
-                inner_to_dttm or to_dttm,
-            )
-            subq = subq.where(and_(*(where_clause_and + [inner_time_filter])))
+            subq = subq.where(and_(*(time_filters + where_clause_and)))
```

Comment:
The `inner_time_filter` logic is there to support the `time_compare` related logic so that the same series exist are used for both queries. Look for `time_compare` in `viz.py` to see how it's used on the caller side...

Reply:
I'll take another look at this.